substrate/frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -427,8 +429,10 @@ pub trait MinerConfig {
 	///
 	/// The weight is computed using `solution_weight`.
 	type MaxWeight: Get<Weight>;
-	/// The maximum number of winners that can be elected.
-	type MaxWinners: Get<u32>;
+	/// The maximum number of winners that can be elected per page (and  overall).
+	type MaxWinnersPerPage: Get<u32>;
+	/// The maximum number of backers (edges) per winner in the last solution.
+	type MaxBackersPerWinner: Get<u32>;
```

Comment:
You need a test to demonstrate that the miner will respect this new configuration now.

Reply:
Deferring this one for now as Unconfirmed issue in AHM https://github.com/paritytech/polkadot-sdk/issues/6136.

Reply:
The issue you have linked seems unrelated :monkey:

Reply:
Let's start from the beginning: What does this configuration mean to the miner?

It means the miner should respect this bound. If it mines a solution that has more than this many backers per winner, it should trim it. 

Then, the question is, is the miner respecting it? the answer seems to be no.

Reply:
Ideally, this needs a phragmen impl that is aware of this constrain, and make a decision based on it. 

otherwise, and short term, if this number is reached, we will sort the backers of a winner, and just remove the excess.

This should not a be a blocker to merge this and use in ahm-next, but it is a blocker before Polkadot deployment.

Reply:
I'm implementing an auto trimming in the phragmen implementation that supports enforcing a max backers per winner: https://github.com/paritytech/polkadot-sdk/pull/6482